tests/unit/io/test_eventletreactor.py

```diff
@@ -18,50 +18,32 @@
 except ImportError:
     import unittest # noqa
 
-from tests.unit.io.utils import submit_and_wait_for_completion, TimerCallback
-from tests import is_eventlet_monkey_patched
+from tests.unit.io.utils import TimerConnectionTests
+from tests.unit.io.eventlet_utils import restore_saved_module
+from tests import notpypy
+from tests import notmonkeypatch
+
 import time
+from eventlet import monkey_patch, kill
 
 try:
     from cassandra.io.eventletreactor import EventletConnection
 except ImportError:
     EventletConnection = None  # noqa
 
-
-class EventletTimerTest(unittest.TestCase):
+@unittest.skipUnless(EventletConnection is not None, "Skpping the eventlet tests because it's not installed")
+@notmonkeypatch
+# There are some issues with some versions of pypy and eventlet
+@notpypy
+class EventletTimerTest(unittest.TestCase, TimerConnectionTests):
 
     def setUp(self):
-        if EventletConnection is None:
-            raise unittest.SkipTest("Eventlet libraries not available")
-        if not is_eventlet_monkey_patched():
-            raise unittest.SkipTest("Can't test eventlet without monkey patching")
+        self.connection_class = EventletConnection
+        # We only to patch the time module
+        monkey_patch(time=True)
```

Comment:
Yeah, I think we should patch everything now, as that's the way the driver is going to be used. I did it when I was hoping I'd be able to unpatch it.
Sorry about that, the reason is in SetUp instead of SetUpClass is because when a class is skipped SetUpClass and tearDownClass are still executed, meaning the patching would occur even if MONKEY_PATCH_LOOP=0